### ISSUE TITLE
Duodecimal input

### ISSUE THREAD

**Issue by Ilmen-vodhr** [Qalculate/qalculate-gtk#135]
Hello,

With earlier versions of Qalculate-gtk I was able to enter duodecimal expressions just fine, but now with the version 3.7.0, I don't know how to enter these anymore without using the ``base()`` function.
When I input ``A`` or ``X`` in duodecimal mode, these are analyzed as unit symbols, and not as digits.
I see that Binary, Octal and Hexadecimal have dedicated prefixes (namely ``0b``, ``0o`` and ``0x``), but apparently duodecimal has none. Maybe ``0d`` could be assigned to it. It might also be desirable to allow inputting expressions using the new Unicode codepoints for the "ten" and "eleven" glyphs for duodecimal, namely ``↊`` and ``↋`` (although I think not many fonts are yet able to render these properly).

Whatever duodecimal input method is available or opted for, it might be desirable to simultaneously allow the ``AB`` notation, the ``XE`` notation and the ``↊↋`` notation as synonymous for the two additional digits.

As a side note, I sometimes privately use an unconventional base notation in which the base is marked with a hash ``#`` prefix preceded by the alphanumeric digit equivalent to the base (e.g. 2 for binary, A for decimal, C for duodecimal, G for hexadecimal…), such that ``2#101`` = 5 in binary, ``A#10`` = ten in decimal, ``C#12`` = 0x12 = eighteen… I find that notation flexible and elegant, and more regular than the ``0x/0o/0b`` notation, but as it's totally unconventional it's very unlikely to be adopted. :p I'm sharing it nevertheless if it can be of any inspiration.
(**Edit:** I vaguely seem to remember there was a programming language with a similar base notation but using a ``r`` (for "radix") instead of the ``#``; however ``r`` could represent a digit in greater bases, it's why I used a hash instead.)

Best regards,
—Ilmen.

**Comment by hanna-kn** [comment]
I have now add support for A and B (case insensitive) for input of duodecimal numbers and "0d" as duodecimal prefix, and expressions containing the duodecimal Unicode characters will now always be interpreted as duodecimal (these characters were already supported when duodecimal input base was selected).

You can use X and E for 10 and 11 in version 3.7.0.

**Comment by Ilmen-vodhr** [comment]
OK, the ``0d`` prefix is a nice addition!
Fiddling a bit more with the interface of version 3.7.0 I just realized most of the radio-button options in the Number Base menu seem to only set the base for the output, and not for the input, so I had to go into "Set Result and Expression Base". As a result I was still having decimal for the input and hadn't noticed it, it's why ``E`` and ``X`` didn't work… Now that I've successfully set duodecimal for the input, everything work fine. I'm sorry for the disturbance. ^^''

I'll take the opportunity to say that I like Qalculate a lot, and to thank you for your great work!
—Ilmen.
